### PR TITLE
[FIX] tools: extract translations in snippets/ directory

### PR DESCRIPTION
**Pull request by mart-e**
Since 14.0, the javascript code of snippets is moved inside a
`/static/src/snippets` directory
Will add missing translations of snippets in `website` and `website_form`

opw-2381030

### CODE REVIEW

**Comment by mart-e** [comment]
@fja-odoo it only contains js and scss files? Never xml?

**Comment by fja-odoo** [comment]
> @fja-odoo it only contains js and scss files? Never xml?

There is xml also yes.

**Comment by fja-odoo** [comment]
@mart-e I think there is also snippets in website_sale, website_mass_mailing, website_event, website_blog, website_mail_channel, website_twitter for the pot file

**Comment by xmo-odoo** [comment]
@mart-e maybe we should just load XML and JS files from `static/src` entire instead of special-casing subdirectories? This issue will rear its head again if e.g. the web team (or some subsets thereof) decide to move to a more widget-based layout or somesuch.

**Comment by mart-e** [comment]
List of subfolders inside `static/src/` for our community and enterprise addons:
- `audio`
- `bugfix`
- `component_hooks`
- `components`
- `css`
- `demo`
- `env`
- `font`
- `fonts`
- `img`
- `js`
- `legacy`
- `less`
- `lib`
- `model`
- `models`
- `owl`
- `pdf`
- `scss`
- `services`
- `snippets`
- `sounds`
- `tests`
- `utils`
- `widgets`
- `xml`
:roll_eyes: 

The `lib` subfolder should definitly be excluded, to see if others are useful

**Comment by xmo-odoo** [comment]
>     * `lib`
> The `lib` subfolder should definitly be excluded, to see if others are useful

third-party libs are supposed to be in static/lib not static/src/lib, so this looks like a fuckup in `website_google_map`.

tests can definitely be excluded though.

**Comment by mart-e** [comment]
cc @alexkuhn since the refactoring of mail client, [most of the translations are missing](https://user-images.githubusercontent.com/564822/99373229-72843c00-28c1-11eb-8a9a-d5fc2d400c70.png) because the code was moved into `components` of `models` folders

**Comment by xmo-odoo** [comment]
At the end of the day I think we can just seek translations in all of src/, possibly moving / fixing the few `tests/` subdirs, I'm not quite sure whether @nim-odoo made a mistake for `website_google_map` or put it there by design to take advantage of a glob or something else.

The rest of the crap is largely assets (images, sounds, fonts, PDFs) which don't belong there but won't be matched by the globs and aren't exactly numerous.

**Comment by alexkuhn** [comment]
@mart-e Few months ago we decided with AL that we can put static code anywhere in `static/src/`, with whatever split of folders we want (there was even some desire to drop `/src`, but there was some things preventing that right now, hence kept for the moment). Folders `components` and `models` will come very often with same structure, i.e. contain "leaf" folders with item files (including tests).

That's why I think it's best to consider translation in whole `static/src/` at least.

**Review by xmo-odoo — APPROVED** [review]

File: odoo/tools/translate.py [hunk 1]
```diff
@@ -1146,17 +1146,17 @@ def _export_translatable_resources(self):
                 for fname in fnmatch.filter(files, '*.py'):
                     self._babel_extract_terms(fname, path, root,
                                               extract_keywords={'_': None, '_lt': None})
-                # Javascript source files in the static/src/js directory, rest is ignored (libs)
-                if fnmatch.fnmatch(root, '*/static/src/js*'):
+                if fnmatch.fnmatch(root, '*/static/src*'):
+                    # Javascript source files in the static/src/js directory
```
Comment (odony) [inline odoo/tools/translate.py:1150]: ... and all over the place :partying_face: 

```suggestion
                    # Javascript source files
```

**Comment by mart-e** [comment]
@robodoo rebase-ff

**Comment by xmo-odoo** [comment]
@mart-e some of the commit messages have typos / seem to have odd grammar:

> Since 14.0, the javascript code of snippets ~~is~~ [has or was] moved inside a
/static/src/snippets directory.

> With owl, the compon**a**nts are defined in static/src/components
Some other modules were using more exotic path and ended up with no
translations.

--

> Only translate the actual strings

Maybe *literal strings* rather than *the actual strings*?

**Comment by mart-e** [comment]
@robodoo r+

**Pull request closed without merging**